### DISCUSSION TITLE
Is there agreed way to get porter manifest file in the mixin?

### DISCUSSION THREAD

**Discussion by dmitriy-kharchenko** [getporter/porter#2787] [Q&A]
Hello folks,

We with colleagues need to pull helm charts to invocation image during its building. 
For this goal it's needed to get a list of charts at build time, but porter passes only data about helm repositories to helm mixin.
For this reason, we need to get the full porter manifest from which we can get information about helm repositories.

So is there some determinated way, where possible to get path to porter manifest from a mixin perspective?

**Comment by alekseybb197** [comment]
Hi, Dmitriy and your colleagues!
It seems to me there is a way to describe the repository in porter.yaml so that during build invovation image porter mixin insert line "RUN helm3 repo update" to dockerfile.
Example https://github.com/getporter/helm3-mixin/blob/master/example/porter.yaml line 4 onwards
Code https://github.com/getporter/helm3-mixin/blob/master/pkg/helm3/build.go line 88 onwards

**Reply by dmitriy-kharchenko** [reply]
Hello Alexey,
Thank you for your answer. Yes, these lines add repositories to invocation image, but I mean charts themselves.
So we can describe charts at install section and they will be downloaded during installation. 
But in airgapped environment (without Internet) there is not possibe to install charts, therefore we become to question: how helm mixin can get full porter manifest for pulling charts during building?

**Reply by alekseybb197** [reply]
I think this can be done with RUN inside dockerfile. Helm mixin only adds lines to the dockerfile. "helm dep update" is called separately with RUN after the lines copying the project inside the image.

**Reply by dmitriy-kharchenko** [reply]
`helm dep update` is called for specific chart. These charts are mentioned in porter manifest and would be cool to get needed invocation image using only one file - porter manifest.
